lib/src/better_command_runner/better_command_runner.dart

```diff
@@ -134,22 +136,29 @@ class BetterCommandRunner extends CommandRunner {
       _onAnalyticsEvent = null;
     }
 
-    await _onBeforeRunCommand?.call(this);
-
-    try {
-      await super.runCommand(topLevelResults);
+    unawaited(Future(() async {
       var command = topLevelResults.command;
-      if (command == null) {
-        _onAnalyticsEvent?.call(BetterCommandRunnerAnalyticsEvents.help);
-      } else {
+      if (command != null) {
         // Command name can only be null for top level results.
         // But since we are taking the name of a command from the top level
         // results there should always be a name specified.
         assert(command.name != null, 'Command name should never be null.');
         _onAnalyticsEvent?.call(
           command.name ?? BetterCommandRunnerAnalyticsEvents.invalid,
         );
+        return;
+      }
+
+      var noUnexpectedArgs = topLevelResults.rest.isEmpty;
+      if (noUnexpectedArgs) {
+        _onAnalyticsEvent?.call(BetterCommandRunnerAnalyticsEvents.help);
       }
```

Comment:
Question: Why was this additional case added? When should it trigger? Do we have a test for it?

Reply:
Yes, several tests started failing when I moved this up 🌟  The reason being that if unrecognized flags are passed the command runner would previously have thrown so this case didn't have to be handled.

What I don't like with this solution is that we have an invisible dependency on the behavior of the `runCommand` method, but I didn't have any better ideas. Any suggestions?